File: C#/0387. First Unique Character in a String.cs
public class Solution {
    public int FirstUniqChar(string s) {
        Dictionary<char,int> sDict=new Dictionary<char,int>();
        foreach(char c in s){
            if(!sDict.ContainsKey(c)){
                sDict[c]=1;
            }
            else{
                sDict[c]+=1;
            }
        }
        int loc=0;
        while(loc<s.Length){
            if(sDict[s[loc]]==1){
                break;
            }
            loc++;
        }
        return loc==s.Length?-1:loc;
    }
}

File: 20190519python/0387. First Unique Character in a String.py
class Solution(object):
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        s_dict={}
        for c in s:
            if c not in s_dict:
                s_dict[c]=1
            else:
                s_dict[c]+=1
        
        for i in range(len(s)):
            if s_dict[s[i]]==1:
                return i
        return -1
